test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/Generators.scala

```diff
@@ -0,0 +1,150 @@
+
+  implicit def arb[A](implicit g: Gen[A]): Arbitrary[A] = Arbitrary(g)
+
+
+}
+
+sealed trait WithGenGenerator {
+
+  def gen[A](implicit arb: Arbitrary[A]): Gen[A] = arb.arbitrary
+
+}
+
+trait SessionDataGen { this: WithGenGenerator =>
+
+  implicit val sessionDataGen: Gen[SessionData] = gen[SessionData]
+
+}
+
+trait BusinessPartnerRecordGen { this: WithGenGenerator =>
+
+  implicit val bprArb: Gen[BusinessPartnerRecord] = gen[BusinessPartnerRecord]
+
+  implicit val bprRequestArb: Gen[BusinessPartnerRecordRequest] = gen[BusinessPartnerRecordRequest]
+
+}
+
+trait IdGen { this: WithGenGenerator =>
+
+  implicit val cgtReferenceArb: Gen[CgtReference] = gen[CgtReference]
+
+  implicit val ggCredIdGen: Gen[GGCredId] = gen[GGCredId]
+
+}
+
+trait NameGen { this: WithGenGenerator =>
+
+  implicit val contactNameGen: Gen[ContactName] = gen[ContactName]
+
+  implicit val individualNameGen: Gen[IndividualName] = gen[IndividualName]
+
+  implicit val trustNameGen: Gen[TrustName] = gen[TrustName]
+
+}
+
+trait JourneyStatusGen { this: WithGenGenerator =>
+
+  implicit val journeyStatusGen: Gen[JourneyStatus] = gen[JourneyStatus]
+
+  implicit val registrationReadyGen: Gen[RegistrationReady] = gen[RegistrationReady]
+
+  implicit val subscriptionReadyGen: Gen[SubscriptionReady] = gen[SubscriptionReady]
+
+  implicit val subscriptionSuccessfulGen: Gen[SubscriptionSuccessful] = gen[SubscriptionSuccessful]
+
+  implicit val subscribedGen: Gen[Subscribed] = gen[Subscribed]
+
+  implicit val individualSupplyingInformationGen: Gen[IndividualSupplyingInformation] =
+    gen[IndividualSupplyingInformation]
+
+  implicit val individualMissingEmailGen: Gen[IndividualMissingEmail] = gen[IndividualMissingEmail]
+
+}
+
+trait AddressGen { this: WithGenGenerator =>
+
+  implicit val addressGen: Gen[Address] = gen[Address]
+
+  implicit val ukAddressGen: Gen[UkAddress] = gen[UkAddress]
+
+}
+
+trait NameMatchGen { this: WithGenGenerator =>
+
+  implicit val trustNameMatchDetailsGen: Gen[TrustNameMatchDetails] = gen[TrustNameMatchDetails]
+
+  implicit val individualNameMatchDetailsGen: Gen[IndividualNameMatchDetails] = gen[IndividualNameMatchDetails]
+
+  implicit val individualUnsuccessfulNameMatchAttemptsGen
+    : Gen[UnsuccessfulNameMatchAttempts[IndividualNameMatchDetails]] =
+    gen[UnsuccessfulNameMatchAttempts[IndividualNameMatchDetails]]
+
+}
+
+trait OnboardingDetailsGen { this: WithGenGenerator =>
+
+  implicit val registrationDetailsArb: Gen[RegistrationDetails] = gen[RegistrationDetails]
+
+  implicit val subscriptionDetailsArb: Gen[SubscriptionDetails] = gen[SubscriptionDetails]
+
+  implicit val subscribedDetailsGen: Gen[SubscribedDetails] = gen[SubscribedDetails]
+
+  implicit val subscribedUpdateDetailsGen: Gen[SubscribedUpdateDetails] = gen[SubscribedUpdateDetails]
+
+}
+
+trait EmailGen { this: WithGenGenerator =>
+
+  implicit val emailGen: Gen[Email] = gen[Email]
+
+  implicit val emailSourceGen: Gen[EmailSource] = gen[EmailSource]
+}
```

Comment:
So now, for every new type for which we need to generate arbitrary values we will need a corresponding entry here?

Reply:
yes, the compilation time is reduced because the work that shapeless has to do to generate each `Arbitrary` instance is now once per type instead of everywhere in each file the instance was needed. It's a bit of overhead to write, but in the long run it's quicker than waiting for tests to compile